sahi/predict.py

```diff
@@ -431,13 +443,14 @@ def predict(
     else:
         image_path_list = [source]
 
-    # init export directories
-    save_dir = Path(increment_path(Path(project) / name, exist_ok=False))  # increment run
-    crop_dir = save_dir / "crops"
-    visual_dir = save_dir / "visuals"
-    visual_with_gt_dir = save_dir / "visuals_with_gt"
-    pickle_dir = save_dir / "pickles"
-    save_dir.mkdir(parents=True, exist_ok=True)  # make dir
+    if export_visual:
```

Comment:
we have to remove this if, they dont depend on `export_visual` param.

save_dir: can be used for coco result json export
crop_dir: is used when export_crop=True`
pickle_dir: is used when `export_pickle=True`

Reply:
I have used this if block because when every run sahi, it was creating null output file in `runs/predict.`  We can remove it, if null output file is unimportant. Or we will add  `if export_visual:` on top of `save_dir.mkdir(parents=True, exist_ok=True)  # make dir`

Reply:
@madenburak this line should be updated as:
```python
if export_visual or export_pickle or export_crop or dataset_json_path is not None:
```

Reply:
> @madenburak this line should be updated as:
> 
> ```python
> if export_visual or export_pickle or export_crop or dataset_json_path is not None:
> ```

I am correcting, now.